SasquatchMac/SasquatchMac/ViewControllers/AppCenterViewController.swift

```diff
@@ -1,168 +1,139 @@
+  @IBOutlet var setEnabledButton : NSButton?
+    
+  @IBOutlet weak var deviceIdField: NSTextField!
+  @IBOutlet weak var startupModeField: NSComboBox!
+  @IBOutlet weak var storageMaxSizeField: NSTextField!
+  @IBOutlet weak var storageFileSizeField: NSTextField!
+  @IBOutlet weak var signInButton: NSButton!
+  @IBOutlet weak var signOutButton: NSButton!
+  @IBOutlet weak var overrideCountryCodeButton: NSButton!
+
+  private var dbFileDescriptor: CInt = 0
+  private var dbFileSource: DispatchSourceProtocol?
+
+  deinit {
+      self.dbFileSource?.cancel()
+      close(self.dbFileDescriptor)
+      UserDefaults.standard.removeObserver(self, forKeyPath: kMSStorageMaxSizeKey)
+  }
+
+  override func viewWillAppear() {
+    setEnabledButton?.state = appCenter.isAppCenterEnabled() ? 1 : 0
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    installIdLabel?.stringValue = appCenter.installId()
+    appSecretLabel?.stringValue = appCenter.appSecret()
+    logURLLabel?.stringValue = appCenter.logUrl()
+    userIdLabel?.stringValue = UserDefaults.standard.string(forKey: kMSUserIdKey) ?? ""
+    setEnabledButton?.state = appCenter.isAppCenterEnabled() ? 1 : 0
+  
+    deviceIdField?.stringValue = AppCenterViewController.getDeviceIdentifier()!
+    let indexNumber = UserDefaults.standard.integer(forKey: kMSStartTargetKey)
+    startupModeField.selectItem(at: indexNumber)
+
+    // Storage size section.
+    storageMaxSizeField.delegate = self
+    let storageMaxSize = UserDefaults.standard.object(forKey: kMSStorageMaxSizeKey) as? Int ?? kMSDefaultDatabaseSize
+    UserDefaults.standard.addObserver(self, forKeyPath: kMSStorageMaxSizeKey, options: .new, context: nil)
+    self.storageMaxSizeField?.stringValue = "\(storageMaxSize / 1024)"
+
+    if let supportDirectory = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first {
+        let dbFile = supportDirectory.appendingPathComponent(Bundle.main.bundleIdentifier!).appendingPathComponent(kMSAppCenterBundleIdentifier).appendingPathComponent("Logs.sqlite")
+        func getFileSize(_ file: URL) -> Int {
+            return (try? file.resourceValues(forKeys:[.fileSizeKey]))?.fileSize ?? 0
+        }
+        self.dbFileDescriptor = dbFile.withUnsafeFileSystemRepresentation { fileSystemPath -> CInt in
+            return open(fileSystemPath!, O_EVTONLY)
+        }
+        self.dbFileSource = DispatchSource.makeFileSystemObjectSource(fileDescriptor: self.dbFileDescriptor, eventMask: [.write], queue: DispatchQueue.main)
+        self.dbFileSource!.setEventHandler {
+            self.storageFileSizeField.stringValue = "\(getFileSize(dbFile) / 1024)"
+        }
+        self.dbFileSource!.resume()
+        self.storageFileSizeField.stringValue = "\(getFileSize(dbFile) / 1024)"
+    }
+  }
+
+  @IBAction func setEnabled(sender : NSButton) {
+    appCenter.setAppCenterEnabled(sender.state == 1)
+    sender.state = appCenter.isAppCenterEnabled() ? 1 : 0

```

Comment:
Can we use the enum values supplied by `UIControlState` instead of "magic" numbers `1` and `0`?